client/signup/steps/domains/index.jsx

```diff
@@ -239,9 +239,6 @@ class DomainsStep extends React.Component {
 	};
 
 	submitWithDomain = ( googleAppsCartItem, shouldHideFreePlan = false ) => {
-		const shouldHideFreePlanItem = this.getIsPlanSelectionUnavailableInFlow()
-			? { shouldHideFreePlan }
-			: {};
 		const shouldUseThemeAnnotation = this.shouldUseThemeAnnotation();
 		const useThemeHeadstartItem = shouldUseThemeAnnotation
 			? { useThemeHeadstart: shouldUseThemeAnnotation }
```

Comment:
The `useThemeHeadstart` is another place that could simply always return the `{ useThemeHeadstart: boolean }` value no matter if `true` or `false`.

Otherwise, as this component is also used by the `domains-theme-preselected` step, that step's `providesDependencies: [ 'useThemeHeadstart' ]` contract is not always honored.